src/app/Caches/Create/ConfigurationFeature.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Divider,
+    Form,
+    FormGroup,
+    Grid,
+    GridItem,
+    InputGroup,
+    Text,
+    TextContent,
+    TextVariants,
+    TextInput,
+    Radio,
+    Select,
+    SelectOption,
+    SelectVariant,
+} from '@patternfly/react-core';
+import { CacheFeature, MaxSizeUnit, EvictionStrategy, EvictionType } from "@services/infinispanRefData";
+import { useTranslation } from 'react-i18next';
+import { MoreInfoTooltip } from '@app/Common/MoreInfoTooltip';
+
+const ConfigurationFeature = (props: {
+    cacheFeature: CacheFeatureStep,
+    cacheFeatureModifier: (CacheFeatureStep) => void,
+    handleIsFormValid: (isFormValid: boolean) => void
+}) => {
+
+    const { t } = useTranslation();
+
+    const [cacheFeatureSelected, setCacheFeatureSelected] = useState(props.cacheFeature.cacheFeatureSelected);
+
+    //Bounded Cache
+    const [evictionType, setEvictionType] = useState<'size' | 'count'>(props.cacheFeature.evictionType);
+    const [maxSize, setMaxSize] = useState<string | undefined>(props.cacheFeature.maxSize);
+    const [maxCount, setMaxCount] = useState<string | undefined>(props.cacheFeature.maxCount);
+    const [evictionStrategy, setEvictionStrategy] = useState<string>(props.cacheFeature.evictionStrategy);
+
+    const [isOpenCacheFeature, setIsOpenCacheFeature] = useState(false);
+    const [isOpenEvictionStrategy, setIsOpenEvictionStrategy] = useState(false);
+    const [isOpenMaxSizeUnit, setIsOpenMaxSizeUnit] = useState(false);
+
+    // Helper states for the maxSize input
+    const [maxSizeNumber, setMaxSizeNumber] = useState<string>();
+    const [maxSizeUnit, setMaxSizeUnit] = useState<MaxSizeUnit>(MaxSizeUnit.MB);
+
+    useEffect(() => {
```

Comment:
@karesti UseEffect here is used multiple times, not sure if it needs refracting.

Reply:
should be ok.